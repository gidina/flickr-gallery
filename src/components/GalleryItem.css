.gallery_item {
    width: 100%;
    height: 100%;
    background: #fff;
    border: 1px solid #f3f0f0;
    box-shadow: 0 1px 2px #afafaf;
    border-radius: 5px;
  }
  
  .gallery_item_preview {
    margin: 0.5em;
    transition: all 0.5s ease;
    position: relative;
    display: inline-block;
    overflow: hidden;
  }
  
  .gallery_item_preview svg.gallery_top {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 50px;
    transition: all 0.5s ease;
  }
  
  .gallery_item_preview button {
    position: absolute;
    top: 0.5em;
    left: -30em;
    background-color: #fff;
    color: #f57171;
    padding: 0.56em 1em;
    border-radius: 5px 0 0 0;
    transition: all 1s ease;
  }
  
  .gallery_item_preview img {
    width: 100%;
    height: 200px;
    border-radius: 5px;
  }
  
  .gallery_item_preview h3 {
    font-family: "Leckerli One", cursive;
  }
  
  .gallery_item_preview p {
    word-wrap: break-word;
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
  }
  
  .gallery_item:hover {
    cursor: pointer;
    color: #777;
    transition: all 0.5s ease;
  }
  
  .gallery_item:hover svg.gallery_top {
    top: -10em;
    transition: all 0.5s ease;
  }
  
  .gallery_item:hover button {
    cursor: pointer;
    left: 0.5em;
    transition: all 1s ease;
  }