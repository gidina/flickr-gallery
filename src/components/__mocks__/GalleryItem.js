import React from "react";
export default class GalleryItem extends React.Component {
    render() {
        return <gallery-item-mock {...this.props}>{this.props.children}</gallery-item-mock>;
    }
}
