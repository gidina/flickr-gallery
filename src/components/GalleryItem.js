import React from "react";
import PropTypes from 'prop-types';
import { primaryColor, secondaryColor } from "../config";
import "./GalleryItem.css";

const GalleryItem = ({ photo, onClick }) => 
    <div className="gallery_item" onClick={onClick}>
        <span className="gallery_item_preview">
            <button>Click to view</button>
            <svg
                fill={photo.media === "video" ? secondaryColor : primaryColor}
                className="gallery_top"
                viewBox="0 3 60 20"
            >
                <path d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z" />
            </svg>
            <img
                src={`https://farm${photo.farm}.staticflickr.com/${photo.server}/${photo.id}_${photo.secret}_b.jpg`}
                alt=""
            />
            <span>
                <h3>{photo.title}</h3>
                <p>{photo.description._content}</p>
            </span>
        </span>
    </div>;

GalleryItem.propTypes = {
    photo: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
};

export default GalleryItem;
