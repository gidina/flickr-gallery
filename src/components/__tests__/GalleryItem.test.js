import React from "react";
import GalleryItem from "../GalleryItem";
import renderer from "react-test-renderer";
import ReactTestUtils from "react-dom/test-utils";

test("Renders correctly with photo properties. If photo.media is 'video', svg must be blue (#5690F7)", () => {
    const photoTest = {
        id: 1,
        title: "Photo 1",
        description: { _content: "Description Photo 1" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 123456
    };
    const tree = renderer
        .create(<GalleryItem photo={photoTest} onClick={() => { }} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Renders correctly with photo properties. If photo.media is different to 'video', svg must be red (#f55)", () => {
    const photoTest = {
        id: 1,
        title: "Photo 1",
        description: { _content: "Description Photo 1" },
        media: "photo",
        farm: 1,
        server: "server1",
        secret: 123456
    };
    const tree = renderer
        .create(<GalleryItem photo={photoTest} onClick={() => { }} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("onClick function is triggered when div with 'gallery_item' class is clicked", () => {
    const onClickMock = jest.fn();
    const photoTest = {
        id: 1,
        title: "Photo 1",
        description: { _content: "Description Photo 1" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 123456
    };

    // Wrapper component for renderIntoDocument functional Component
    class Wrapper extends React.Component {
        render() {
            return this.props.children
        }
    }

    const tree = ReactTestUtils.renderIntoDocument(<Wrapper><GalleryItem photo={photoTest} onClick={onClickMock} /></Wrapper>);
    expect(tree.props.children.props.onClick).toEqual(onClickMock);
});
