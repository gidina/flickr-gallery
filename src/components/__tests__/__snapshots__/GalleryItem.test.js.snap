// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with photo properties. If photo.media is 'video', svg must be blue (#5690F7) 1`] = `
<div
  className="gallery_item"
  onClick={[Function]}
>
  <span
    className="gallery_item_preview"
  >
    <button>
      Click to view
    </button>
    <svg
      className="gallery_top"
      fill="#5690F7"
      viewBox="0 3 60 20"
    >
      <path
        d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z"
      />
    </svg>
    <img
      alt=""
      src="https://farm1.staticflickr.com/server1/1_123456_b.jpg"
    />
    <span>
      <h3>
        Photo 1
      </h3>
      <p>
        Description Photo 1
      </p>
    </span>
  </span>
</div>
`;

exports[`Renders correctly with photo properties. If photo.media is different to 'video', svg must be red (#f55) 1`] = `
<div
  className="gallery_item"
  onClick={[Function]}
>
  <span
    className="gallery_item_preview"
  >
    <button>
      Click to view
    </button>
    <svg
      className="gallery_top"
      fill="#f55"
      viewBox="0 3 60 20"
    >
      <path
        d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z"
      />
    </svg>
    <img
      alt=""
      src="https://farm1.staticflickr.com/server1/1_123456_b.jpg"
    />
    <span>
      <h3>
        Photo 1
      </h3>
      <p>
        Description Photo 1
      </p>
    </span>
  </span>
</div>
`;
